src/nvim/option.c

```diff
@@ -4195,6 +4255,21 @@ set_num_option (
 
   options[opt_idx].flags |= P_WAS_SET;
 
+  if (!starting && errmsg == NULL) {
+    char_u buf_old[11], buf_new[11], buf_type[7];
+    vim_snprintf((char *)buf_old, 10, "%ld", old_value);
```

Comment:
@oni-link I would actually suggest using `NUMBUFLEN` constant. I have no idea why it is defined to `30` (for floating-point numbers? I was unable to make them occupy more then fourteen characters with `string()`), but this is standard constant for use for buffers which are going to hold stringified numbers.